### PR TITLE
Writing stories section migration for 6 0

### PR DESCRIPTION
**Pull request by jonniebigodes**
This pull request adds the writing stories section to the documentation based on the requirements.

What was done:

- The content was added based on the existing requirements.
- Images will be added when they are available with further commits.
- Some items in the documentation are noted as they need to be better fined tuned during the review process.

Feel free to provide feedback

### CODE REVIEW

File: docs/writing-stories/introduction.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+The default export metadata controls how Storybook lists your stories and provides information used by addons. For example, here’s the default export for a story file `Button.stories.js`:
+
+```js
+// Button.stories.js
+
+import { Button } from './Button';
+
+export default {
+  title: ‘Components/Button’,
```
Comment (tmeasday) [inline docs/writing-stories/introduction.md:34]:
```suggestion
  title: 'Components/Button',
```

Looks like some quotes have gotten messed up

File: docs/writing-stories/introduction.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+import Button from ‘./Button’;
+export {
+  title: ‘Button’,
+  component: Button,
+  decorators: [(Story) => <div style={{ padding: ‘3em’ }}><Story /></div>]
+}
+```
+<div style="background-color:#F8FAFC">
+TODO: a little more clarification on where these two links ("can be more complex" and "addons") below will point to
```
Comment (tmeasday) [inline docs/writing-stories/introduction.md:171]: complex decorators: https://docs.google.com/document/d/1BawdfqGhsP5tqhYQbVn29c6_rAz1ePmk9QayoXLDOKM/edit#heading=h.hh34swttoanz

addons: https://docs.google.com/document/d/1BawdfqGhsP5tqhYQbVn29c6_rAz1ePmk9QayoXLDOKM/edit#heading=h.klqpukxeadji

File: docs/writing-stories/args.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+```js
+export const Primary = Story.bind({});
+Primary.args = {
+  primary: true,
+  label: ‘Primary’,
+}
+```
+
+<div style="background-color:#F8FAFC">
+TODO: ask dom/tom for clarity on the reuse link location
```
Comment (tmeasday) [inline docs/writing-stories/args.md:30]: I think I was thinking of this: https://docs.google.com/document/d/1BawdfqGhsP5tqhYQbVn29c6_rAz1ePmk9QayoXLDOKM/edit#heading=h.mjsq7rpmc8oq

File: docs/writing-stories/args.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+### Using args in addons
+
+If you are [writing an addon](../api/addons#getting-started) that wants to read or update args, use the `useArgs` hook exported by `@storybook/api`:
+
+```js
+import { useArgs } from ‘@storybook/api’;
+const [args, updateArgs] = useArgs();
+// To update one or more args:
+updateArgs({ key: ‘value’ });
```
Comment (tmeasday) [inline docs/writing-stories/args.md:114]:
```suggestion
import { useArgs } from ‘@storybook/api’;
const [args, updateArgs, resetArgs] = useArgs();
// To update one or more args:
updateArgs({ key: ‘value’ });
// To reset one (or more) args:
resetArgs(argNames: ['key']);
// To reset all args:
resetArgs();
```

Comment (tmeasday) [reply docs/writing-stories/args.md:114]: This API got extended a little.

File: docs/writing-stories/parameters.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+For example, let’s customize the backgrounds addon via a parameter. We’ll use `parameters.backgrounds` to define which backgrounds appear in the backgrounds toolbar when a story is selected.
+
+### Story parameters
+
+We can set a parameter for a single story with the `parameters` key on a CSF export:
+
+```js
+export const Primary = Story.bind({});
+Primary.args = …
+Primary.parameters = { backgrounds: … };
```
Comment (tmeasday) [inline docs/writing-stories/parameters.md:16]: We should probably put legitimate values for the backgrounds addon in here.

File: docs/writing-stories/parameters.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+### Rules of parameter inheritance
+
+The way the global, component and story parameters are combined is:
+
+- More specific parameters take precedence (so a story parameter overwrites a component parameter which overwrites a global parameter).
+- Parameters are **merged** so keys are only ever overwritten, never dropped.
+
+The merging of parameters is important. It means it is possible to override a single specific sub-parameter on a per-story basis but still retain the majority of the parameters defined globally.
+
+If you are defining an API that relies on parameters (e..g an __addon__) it is a good idea to take this behaviour into account.
```
Comment (tmeasday) [inline docs/writing-stories/parameters.md:59]: Link to writing addons

File: docs/writing-stories/decorators.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+<div style="background-color:#F8FAFC">
+TODO: per screenshot spreadsheet add Image of Component Story with margins
+</div>
+
+
+### “Context” for mocking
+
+Some libraries require components higher up in the component hierarchy to render properly. For example in Styled Components, a `ThemeProvider` is required if your components make use of themes. Add a single global decorator that add this context to to all stories in __preview.js__ :
```
Comment (tmeasday) [inline docs/writing-stories/decorators.md:31]: We should always spell out `.storybook/preview.js` and link to the docs (people won't know what it is, necessarily).

File: docs/writing-stories/naming-components-and-hierarchy.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    storySort: {
+      method: 'alphabetical', // Optional, defaults to 'configure'.
+      order: ['Intro', 'Components'], // Optional, defaults to [].
+      locales: 'en-US', // Optional, defaults to system locale.
+    },
+  },
+};
+```
+<div>
+TODO: ask tom/dom if we could add table here that could be further expanded
```
Comment (tmeasday) [inline docs/writing-stories/naming-components-and-hierarchy.md:87]: Seems like a good idea.

**Review by tmeasday — COMMENTED** [review]
Looking good, some more notes.

File: docs/writing-stories/introduction.md (hunk 1 above)
Comment (jonniebigodes) [reply docs/writing-stories/introduction.md]: i've taken care of them in the get-started section and now here

File: docs/writing-stories/introduction.md (hunk 2 above)
Comment (jonniebigodes) [reply docs/writing-stories/introduction.md]: links were updated per feedback.

File: docs/writing-stories/args.md (hunk 3 above)
Comment (jonniebigodes) [reply docs/writing-stories/args.md]: the link was updated per feedback

File: docs/writing-stories/args.md (hunk 4 above)
Comment (jonniebigodes) [reply docs/writing-stories/args.md]: the snippet was updated per feedback

File: docs/writing-stories/parameters.md (hunk 6 above)
Comment (jonniebigodes) [reply docs/writing-stories/parameters.md]: the link is now added.

File: docs/writing-stories/naming-components-and-hierarchy.md (hunk 8 above)
Comment (jonniebigodes) [reply docs/writing-stories/naming-components-and-hierarchy.md]: i've just committed the table with a simple structure that we can probably use. Take a look at it and let me know

File: docs/writing-stories/decorators.md (hunk 7 above)
Comment (jonniebigodes) [reply docs/writing-stories/decorators.md]: Just updated the wording. The link is now present,

File: docs/writing-stories/parameters.md (hunk 5 above)
Comment (jonniebigodes) [reply docs/writing-stories/parameters.md]: let me know if you're ok with the new update on this snippet. I've updated it recently

Comment (tmeasday) [reply docs/writing-stories/parameters.md]: LGTM!

File: docs/writing-stories/naming-components-and-hierarchy.md (hunk 8 above)
Comment (tmeasday) [reply docs/writing-stories/naming-components-and-hierarchy.md]: Looks good!

File: docs/writing-stories/args.md [hunk 9]
```diff
@@ -0,0 +1,136 @@
+---
+title: 'Args'
+---
+
+A story is a component with a set of arguments (props, slots, inputs, etc). “Args” are Storybook’s mechanism for defining those arguments as a first class entity that’s machine readable. This allows Storybook and its addons to live edit components. You do not need to change your underlying component code to use args.
```
Comment (domyen) [inline docs/writing-stories/args.md:5]:
```suggestion
A story is a component with a set of arguments (props, slots, inputs, etc). “Args” are Storybook’s mechanism for defining those arguments as a first class entity that’s machine readable. This allows Storybook and its addons to live edit components. You *do not* need to change your underlying component code to use args.
```

File: docs/writing-stories/args.md [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+When an arg’s value is changed, the component re-renders, allowing you to interact with components in Storybook’s UI via addons that affect args.
+
+
+For a discussion on how and why to write stories with args, check the [using args](./introduction#using-args) section. For details on how args work, read on.
```
Comment (domyen) [inline docs/writing-stories/args.md:10]:
```suggestion
Learn how and why to write stories with args [here](./introduction#using-args) section. For details on how args work, read on.
```

File: docs/writing-stories/decorators.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+### “Context” for mocking
+
+Some libraries require components higher up in the component hierarchy to render properly. For example in Styled Components, a `ThemeProvider` is required if your components make use of themes. Add a single global decorator that add this context to to all stories in [`.storybook/preview.js`](../configure/overview#configure-story-rendering):
+
+```js
+// .storybook/preview.js
+
+import { ThemeProvider } from 'styled-components';
+export const decorators = [
+  (Story) => <ThemeProvider theme=”default”><Story/></ThemeProvider>,
```
Comment (domyen) [inline docs/writing-stories/decorators.md:35]:
```suggestion
  (Story) => <ThemeProvider theme="default"><Story/></ThemeProvider>,
```

File: docs/writing-stories/decorators.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+];
+```
+
+In the example above, the theme is hardcoded to a mock value of `”default”` but you may want to vary that value, either on a per-story basis (if it is data you are mocking that is relevant to the other args of the story) or in a user controlled way (for instance to provide a theme switcher).
```
Comment (domyen) [inline docs/writing-stories/decorators.md:39]:
```suggestion
In the example above, the theme is hardcoded to a mock value of `default` but you may want to vary that value, either on a per-story basis (if it is data you are mocking that is relevant to the other args of the story) or in a user controlled way (for instance to provide a theme switcher).
```

File: docs/writing-stories/decorators.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+The second argument to a decorator function is the **story context** which in particular contains the keys:
+
+- `args` - the story arguments. You can use some args in your decorators and drop them in the story implementation itself.
+
+- `globals` - the Storybook-wide [globals](../essentials/toolbars-and-globals#globals). In particular you can use the [toolbars feature](../essentials/toolbars-and-globals#global-types-toolbar-annotations) to allow you to change these values using Storybook’s UI.
+
+
+#### Using decorators to provide data
+
+If your components are [“connected”] and require side-loaded data to render, you can use decorators to provide that data in a mocked way, without having to refactor your components to take that data as an arg. There are several techniques to achieve this, depending on exactly how you are loading that data -- read more in the [building pages in Storybook](../workflows/build-pages-with-storybook) section.
```
Comment (domyen) [inline docs/writing-stories/decorators.md:50]:
```suggestion
If your components are “connected” and require side-loaded data to render, you can use decorators to provide that data in a mocked way, without having to refactor your components to take that data as an arg. There are several techniques to achieve this, depending on exactly how you are loading that data -- read more in the [building pages in Storybook](../workflows/build-pages-with-storybook) section.
```

File: docs/writing-stories/introduction.md [hunk 14]
```diff
@@ -0,0 +1,235 @@
...
+A component’s stories are defined in a story file that lives alongside the component file. The story file is for development-only, it won't be included in your production bundle.
+
+```
+Button.js | ts
+Button.stories.js | ts
+```
+
+### Component Story Format
+
+We define stories according to the Component Story Format (CSF), an ES6 module-based standard that is portable between tools and easy to write.
```
Comment (domyen) [inline docs/writing-stories/introduction.md:20]: Add link to the CSF page

File: docs/writing-stories/introduction.md [hunk 15]
```diff
@@ -0,0 +1,235 @@
...
+```
+Button.js | ts
+Button.stories.js | ts
+```
+
+### Component Story Format
+
+We define stories according to the Component Story Format (CSF), an ES6 module-based standard that is portable between tools and easy to write. 
+
+The key ingredients are the **`default` export** that describes the component, and **named exports** that describe the stories.
```
Comment (domyen) [inline docs/writing-stories/introduction.md:22]: Add links to MDN docs for default exports and named exports respectively. People sometimes have trouble discerning the difference.

File: docs/writing-stories/introduction.md [hunk 16]
```diff
@@ -0,0 +1,235 @@
...
+export const Secondary = Template.bind({});
+Secondary.args = {  ...Primary.args,  label: '😄👍😍💯',};
+
+export const Tertiary = Template.bind({});
+Tertiary.args = {  ...Primary.args,  label: '📚📕📈🤓',};
+```
+
+The template is reused across stories. Template.bind({}) makes a copy of the function which reduces code duplication. Similarly,`...Primary.args` makes a copy of the data, reducing data duplication.
+
+What’s more, you can import args to reuse when writing stories for other components. This is useful when you’re building composite components. For example, if we make a `ButtonGroup`component, we might remix two `Button` stories.
```
Comment (domyen) [inline docs/writing-stories/introduction.md:90]:
```suggestion
What’s more, you can import args to reuse when writing stories for other components. This is useful when you’re building composite components. For example, if we make a `ButtonGroup` story, we might remix two stories from its child component `Button`.
```

File: docs/writing-stories/naming-components-and-hierarchy.md [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+Yields this:
+
+![Stories hierarchy without paths](./naming-hierarchy-no-path.png)
+
+
+### Grouping
+
+It is also possible to group related components in an expandable interface in order to help with Storybook organization. To do so, use the `’/’` as a separator:
```
Comment (domyen) [inline docs/writing-stories/naming-components-and-hierarchy.md:22]:
```suggestion
It is also possible to group related components in an expandable interface in order to help with Storybook organization. To do so, use the `/` as a separator:
```

**Review by domyen — CHANGES_REQUESTED** [review]

File: docs/writing-stories/parameters.md [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+
+### Component parameters
+
+We can set the parameters for all stories of a component using the `parameters` key on the default CSF export:
+
+```js
+import Button from './';
+
+export default {
+  title: “Button”,
```
Comment (domyen) [inline docs/writing-stories/parameters.md:37]:
```suggestion
  title: "Button",
```

File: docs/writing-stories/args.md (hunk 9 above)
Comment (jonniebigodes) [reply docs/writing-stories/args.md]: Fixed with latest commit

File: docs/writing-stories/decorators.md (hunk 11 above)
Comment (jonniebigodes) [reply docs/writing-stories/decorators.md]: Fixed with latest commit

File: docs/writing-stories/decorators.md (hunk 12 above)
Comment (jonniebigodes) [reply docs/writing-stories/decorators.md]: Fixed with latest commit

File: docs/writing-stories/decorators.md (hunk 13 above)
Comment (jonniebigodes) [reply docs/writing-stories/decorators.md]: Fixed with latest commit

File: docs/writing-stories/introduction.md (hunk 16 above)
Comment (jonniebigodes) [reply docs/writing-stories/introduction.md]: Fixed with latest commit

File: docs/writing-stories/naming-components-and-hierarchy.md (hunk 17 above)
Comment (jonniebigodes) [reply docs/writing-stories/naming-components-and-hierarchy.md]: Fixed with latest commit

File: docs/writing-stories/parameters.md (hunk 18 above)
Comment (jonniebigodes) [reply docs/writing-stories/parameters.md]: Fixed with latest commit

File: docs/writing-stories/introduction.md (hunk 15 above)
Comment (jonniebigodes) [reply docs/writing-stories/introduction.md]: Links are added.

File: docs/writing-stories/introduction.md (hunk 14 above)
Comment (jonniebigodes) [reply docs/writing-stories/introduction.md]: Fixed with latest commit

File: docs/writing-stories/args.md (hunk 10 above)
Comment (jonniebigodes) [reply docs/writing-stories/args.md]: Fixed with latest commit

**Review by domyen — APPROVED** [review]
lgtm 🎉🎉🎉🎉

**Comment by jonniebigodes** [comment]
Merging it as is, follow up changes will be made into subsquent pull requests.

**Pull request merged**